<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Linked Charts for Single Cells • lcsc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Linked Charts for Single Cells">
<meta property="og:description" content="Classification of single cells based on nearest-neighbor smoothing which does not rely on unsupervised learning methods.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">lcsc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/covid-balf-study.html">Analyzing COVID-19 BALF samples</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="lcsc-linked-charts-for-single-cells" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#lcsc-linked-charts-for-single-cells" class="anchor"></a>lcsc (linked charts for single cells)</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<div id="what-is-it-about" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-it-about" class="anchor"></a>What is it about?</h2>
<p>This package allows you to classify single cells based on nearest-neighbor smoothing, not relying on unsupervised learning methods. The documentation can be found <a href="https://anders-biostat.github.io/lcsc/index.html">here</a>.</p>
<p>Most often annotation of single cell data is done by clustering via graph-based methods followed by differential gene expression analysis between the clusters to find certain marker genes.</p>
<p>In this package we propose to directly start with the marker gene and calculate a smoothed fraction for the marker gene and to select cells based on whether they are above or below a chosen threshold. This smoothed fraction is calculated over the <code>k</code> nearest neighbors by dividing the sum of the marker gene counts for the nearest neighbors by their sum of total counts (“library size”). This fraction is subsequently raised to the power of <span class="math inline"><em>γ</em></span> as some sort of gamma-correction (or Box-Cox correction). The threshold can then be simply chosen by looking at the distribution of this smoothed expression fraction. A screenshot is provided below.</p>
<p><img src="reference/figures/formula.png" width="40%" style="display: block; margin: auto;"></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install the development version from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"anders-biostat/lc-sc"</span><span class="op">)</span></code></pre></div>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>As in the <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">Guided Clustering Tutorial by Seurat</a>, we will be using the PBMCdataset from 10X containing 2,700 single cells that were sequenced on the Illumina NextSeq 500. The raw data are made availbe by 10X <a href="https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz">here</a>.</p>
<p>We start of with the standard workflow consisting of quality control, normalization, and linear/non-linear dimensional reduction using <code>Seurat</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>

<span class="va">pbmc.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"inst/extdata/filtered_gene_bc_matrices/hg19/"</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/CreateSeuratObject.html">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pbmc.data</span>, project <span class="op">=</span> <span class="st">"pbmc3k"</span>, min.cells <span class="op">=</span> <span class="fl">3</span>, min.features <span class="op">=</span> <span class="fl">200</span><span class="op">)</span>
<span class="va">pbmc</span><span class="op">[[</span><span class="st">"percent.mt"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">pbmc</span>, pattern <span class="op">=</span> <span class="st">"^MT-"</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">pbmc</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html">ScaleData</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">)</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html">RunPCA</a></span><span class="op">(</span><span class="va">pbmc</span>, features <span class="op">=</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">pbmc</span><span class="op">)</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html">FindNeighbors</a></span><span class="op">(</span><span class="va">pbmc</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>
<span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html">RunUMAP</a></span><span class="op">(</span><span class="va">pbmc</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>
<span class="va">pbmc</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 13714 features across 2700 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (13714 features, 2000 variable features)</span>
<span class="co">#&gt;  2 dimensional reductions calculated: pca, umap</span></code></pre></div>
<p>To run the linked-charts for single cells, we need to extract the following data for the <code>Seurat</code> object.</p>
<ol>
<li>Sparse count matrix (rows = genes, cols = cells)</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="st">"counts"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>
<span class="co">#&gt; [1] 13714  2700</span></code></pre></div>
<ol>
<li>PC coordinates</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pc_space</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Embeddings.html">Embeddings</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="st">"pca"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">pc_space</span><span class="op">)</span>
<span class="co">#&gt; [1] 2700   50</span></code></pre></div>
<ol>
<li>Non-linear dimensional reduction embedding (2D)</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Embeddings.html">Embeddings</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="st">"umap"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">embedding</span><span class="op">)</span>
<span class="co">#&gt; [1] 2700    2</span></code></pre></div>
<ol>
<li>Cell meta data</li>
</ol>
<!-- end list --><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta_data</span> <span class="op">&lt;-</span> <span class="va">pbmc</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">)</span>
<span class="co">#&gt;                  orig.ident nCount_RNA nFeature_RNA percent.mt</span>
<span class="co">#&gt; AAACATACAACCAC-1     pbmc3k       2419          779  3.0177759</span>
<span class="co">#&gt; AAACATTGAGCTAC-1     pbmc3k       4903         1352  3.7935958</span>
<span class="co">#&gt; AAACATTGATCAGC-1     pbmc3k       3147         1129  0.8897363</span>
<span class="co">#&gt; AAACCGTGCTTCCG-1     pbmc3k       2639          960  1.7430845</span>
<span class="co">#&gt; AAACCGTGTATGCG-1     pbmc3k        980          521  1.2244898</span>
<span class="co">#&gt; AAACGCACTGGTAC-1     pbmc3k       2163          781  1.6643551</span></code></pre></div>
<p>Based on these meta we will generate the <code>cells</code> that only contains the needed information and will be used to subset the counts according to which sample is selected.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s</span> <span class="op">=</span> <span class="st">"orig.ident"</span>  <span class="co"># name of the sample column</span>
<span class="va">cells</span> <span class="op">=</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">)</span>,
  sample <span class="op">=</span> <span class="va">meta_data</span><span class="op">[[</span><span class="va">s</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cells</span><span class="op">)</span>
<span class="co">#&gt; Registered S3 method overwritten by 'cli':</span>
<span class="co">#&gt;   method     from         </span>
<span class="co">#&gt;   print.boxx spatstat.geom</span>
<span class="co">#&gt; # A tibble: 6 × 2</span>
<span class="co">#&gt;   id               sample</span>
<span class="co">#&gt;   &lt;chr&gt;            &lt;fct&gt; </span>
<span class="co">#&gt; 1 AAACATACAACCAC-1 pbmc3k</span>
<span class="co">#&gt; 2 AAACATTGAGCTAC-1 pbmc3k</span>
<span class="co">#&gt; 3 AAACATTGATCAGC-1 pbmc3k</span>
<span class="co">#&gt; 4 AAACCGTGCTTCCG-1 pbmc3k</span>
<span class="co">#&gt; 5 AAACCGTGTATGCG-1 pbmc3k</span>
<span class="co">#&gt; 6 AAACGCACTGGTAC-1 pbmc3k</span></code></pre></div>
<p>Additionally we need to generate the neighborhood graph per sample containing the <code>k</code> nearest neighbors for each cell. For the computation the <code>lcsc</code> package provides the <code>run_nn</code> function, which also takes the number of pc_dimensions (<code>dim</code>) to be considered as input.</p>
<p>Start the <code>linked charts</code> application after generating a nearest neighborhood graph per sample.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">lcsc</span><span class="op">)</span>

<span class="va">k</span> <span class="op">=</span> <span class="fl">50</span>
<span class="va">nn</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/run_nn.html">run_nn</a></span><span class="op">(</span><span class="va">cells</span>, <span class="va">pc_space</span>, k<span class="op">=</span><span class="va">k</span>, dim<span class="op">=</span><span class="fl">30</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">nn</span><span class="op">)</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ idx  : num [1:2700, 1:50] 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="co">#&gt;  $ dists: num [1:2700, 1:50] 0 0 0 0 0 0 0 0 0 0 ...</span></code></pre></div>
<p>Now we can finally starts the <code>linked charts</code> application. <code>k</code> refers to the number of nearest neighbors which are used for smoothing. See the equation above.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/lc_vis.html">lc_vis</a></span><span class="op">(</span>cells<span class="op">=</span><span class="va">cells</span>,
       counts<span class="op">=</span><span class="va">counts</span>,
       pc_space<span class="op">=</span><span class="va">pc_space</span>,
       embedding<span class="op">=</span><span class="va">embedding</span>,
       nn<span class="op">=</span><span class="va">nn</span>,
       k<span class="op">=</span><span class="fl">50</span> <span class="co"># Smoothing the expression over 50 nearest neighbors</span>
       <span class="op">)</span></code></pre></div>
<p>The application would look like this after selecting macrophages using the smoothed expression of CD68.</p>
<p><img src="reference/figures/pbmc_lcsc_application.png"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 18.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] lcsc_0.0.0.9000    SeuratObject_4.0.2 Seurat_4.0.4       dplyr_1.0.7       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] nlme_3.1-149          spatstat.sparse_2.0-0 matrixStats_0.60.1   </span>
<span class="co">#&gt;   [4] RcppAnnoy_0.0.19      RColorBrewer_1.1-2    httr_1.4.2           </span>
<span class="co">#&gt;   [7] sctransform_0.3.2     tools_4.0.3           utf8_1.2.2           </span>
<span class="co">#&gt;  [10] R6_2.5.1              irlba_2.3.3           rpart_4.1-15         </span>
<span class="co">#&gt;  [13] KernSmooth_2.23-17    uwot_0.1.10           mgcv_1.8-33          </span>
<span class="co">#&gt;  [16] DBI_1.1.1             lazyeval_0.2.2        colorspace_2.0-2     </span>
<span class="co">#&gt;  [19] tidyselect_1.1.1      gridExtra_2.3         compiler_4.0.3       </span>
<span class="co">#&gt;  [22] cli_3.0.1             plotly_4.9.4.1        scales_1.1.1         </span>
<span class="co">#&gt;  [25] spatstat.data_2.1-0   lmtest_0.9-38         ggridges_0.5.3       </span>
<span class="co">#&gt;  [28] pbapply_1.4-3         goftest_1.2-2         stringr_1.4.0        </span>
<span class="co">#&gt;  [31] digest_0.6.27         spatstat.utils_2.2-0  rmarkdown_2.10       </span>
<span class="co">#&gt;  [34] pkgconfig_2.0.3       htmltools_0.5.1.1     parallelly_1.27.0    </span>
<span class="co">#&gt;  [37] fastmap_1.1.0         highr_0.9             htmlwidgets_1.5.3    </span>
<span class="co">#&gt;  [40] rlang_0.4.11          rstudioapi_0.13       shiny_1.6.0          </span>
<span class="co">#&gt;  [43] generics_0.1.0        zoo_1.8-9             jsonlite_1.7.2       </span>
<span class="co">#&gt;  [46] ica_1.0-2             magrittr_2.0.1        patchwork_1.1.1      </span>
<span class="co">#&gt;  [49] Matrix_1.3-4          Rcpp_1.0.7            munsell_0.5.0        </span>
<span class="co">#&gt;  [52] fansi_0.5.0           abind_1.4-5           reticulate_1.20      </span>
<span class="co">#&gt;  [55] lifecycle_1.0.0       stringi_1.7.3         yaml_2.2.1           </span>
<span class="co">#&gt;  [58] MASS_7.3-53           Rtsne_0.15            plyr_1.8.6           </span>
<span class="co">#&gt;  [61] grid_4.0.3            parallel_4.0.3        listenv_0.8.0        </span>
<span class="co">#&gt;  [64] promises_1.2.0.1      ggrepel_0.9.1         crayon_1.4.1         </span>
<span class="co">#&gt;  [67] deldir_0.2-10         miniUI_0.1.1.1        lattice_0.20-41      </span>
<span class="co">#&gt;  [70] cowplot_1.1.1         splines_4.0.3         tensor_1.5           </span>
<span class="co">#&gt;  [73] knitr_1.33            pillar_1.6.2          igraph_1.2.6         </span>
<span class="co">#&gt;  [76] spatstat.geom_2.2-2   future.apply_1.8.1    reshape2_1.4.4       </span>
<span class="co">#&gt;  [79] codetools_0.2-16      leiden_0.3.9          glue_1.4.2           </span>
<span class="co">#&gt;  [82] evaluate_0.14         data.table_1.14.0     png_0.1-7            </span>
<span class="co">#&gt;  [85] vctrs_0.3.8           httpuv_1.6.2          polyclip_1.10-0      </span>
<span class="co">#&gt;  [88] spatstat.core_2.3-0   gtable_0.3.0          RANN_2.6.1           </span>
<span class="co">#&gt;  [91] purrr_0.3.4           tidyr_1.1.3           scattermore_0.7      </span>
<span class="co">#&gt;  [94] future_1.22.1         assertthat_0.2.1      ggplot2_3.3.5        </span>
<span class="co">#&gt;  [97] xfun_0.25             mime_0.11             xtable_1.8-4         </span>
<span class="co">#&gt; [100] RSpectra_0.16-0       later_1.3.0           survival_3.2-7       </span>
<span class="co">#&gt; [103] viridisLite_0.4.0     tibble_3.1.4          cluster_2.1.0        </span>
<span class="co">#&gt; [106] globals_0.14.0        fitdistrplus_1.1-5    ellipsis_0.3.2       </span>
<span class="co">#&gt; [109] ROCR_1.0-11</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Philipp SL Schäfer <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Philipp SL Schäfer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
