<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing COVID-19 BALF samples • lcsc</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing COVID-19 BALF samples">
<meta property="og:description" content="lcsc">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">lcsc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/covid-balf-study.html">Analyzing COVID-19 BALF samples</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyzing COVID-19 BALF samples</h1>
            
      
      
      <div class="hidden name"><code>covid-balf-study.Rmd</code></div>

    </div>

    
    
<p>This vignette shows an example how <code>lcsc</code> can be used for real data analyses.</p>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load Data</h2>
<p>For demonstration purposes we will be using the dataset created in the study “Single-cell landscape of bronchoalveolar immune cells in COVID-19 patients” by Liao M. et al. In this study, immune cells were isolated and sequenced from Bronchoalveolar Lavage Fluid (BALF) which was taken from healthy subjects and patients diagnosed with COVID-19.</p>
<p>The authors kindly provided the processed data as <code>Seurat</code> object which can be downloaded <a href="http://cells.ucsc.edu/covid19-balf/nCoV.rds">here</a>. Otherwise we could simply run the standard preprocessing steps.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Download</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"http://cells.ucsc.edu/covid19-balf/nCoV.rds"</span>, <span class="st">"nCoV.rds"</span><span class="op">)</span>
<span class="co"># Or directly load into R (not recommended, size is about 3G GB)</span>
<span class="va">balf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="st">"http://cells.ucsc.edu/covid19-balf/nCoV.rds"</span>,<span class="st">"rb"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="va">balf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"../data/nCoV.rds"</span><span class="op">)</span>
<span class="va">balf</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 25916 features across 66452 samples within 2 assays </span>
<span class="co">#&gt; Active assay: RNA (23916 features, 2000 variable features)</span>
<span class="co">#&gt;  1 other assay present: integrated</span>
<span class="co">#&gt;  3 dimensional reductions calculated: pca, tsne, umap</span></code></pre></div>
<p>To run lcsc we need to extract the following data.</p>
<ol style="list-style-type: decimal">
<li>Sparse count matrix</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/AssayData.html">GetAssayData</a></span><span class="op">(</span><span class="va">balf</span>, <span class="st">"counts"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>
<span class="co">#&gt; [1] 23916 66452</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>PC embedding</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pc_space</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Embeddings.html">Embeddings</a></span><span class="op">(</span><span class="va">balf</span>, <span class="st">"pca"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">pc_space</span><span class="op">)</span>
<span class="co">#&gt; [1] 66452   100</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Non-linear dimensional reduction embedding (two-dimensional)</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">embedding</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Embeddings.html">Embeddings</a></span><span class="op">(</span><span class="va">balf</span>, <span class="st">"umap"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">embedding</span><span class="op">)</span>
<span class="co">#&gt; [1] 66452     2</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Cell meta data (“coldata”)</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta_data</span> <span class="op">&lt;-</span> <span class="va">balf</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"sample_new"</span>, <span class="st">"disease"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;                    sample sample_new disease</span>
<span class="co">#&gt; AAACCTGAGACACTAA_1    C51        HC1       N</span>
<span class="co">#&gt; AAACCTGAGGAGTACC_1    C51        HC1       N</span>
<span class="co">#&gt; AAACCTGAGGATATAC_1    C51        HC1       N</span>
<span class="co">#&gt; AAACCTGAGGTCATCT_1    C51        HC1       N</span>
<span class="co">#&gt; AAACCTGCACGGATAG_1    C51        HC1       N</span>
<span class="co">#&gt; AAACCTGCAGGGAGAG_1    C51        HC1       N</span></code></pre></div>
<p>Based on these meta we will generate the <code>cells</code> that only contains the needed information and will be used to subset the counts according to which sample is selected.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sample_column</span> <span class="op">&lt;-</span> <span class="st">"sample_new"</span>   <span class="co"># column for sample name</span>
<span class="va">disease_column</span> <span class="op">&lt;-</span> <span class="st">"disease"</span>   <span class="co"># column for disease state</span>
  
<span class="va">cells</span> <span class="op">=</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
  id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">)</span>,
  sample <span class="op">=</span> <span class="va">meta_data</span><span class="op">[[</span><span class="va">sample_column</span><span class="op">]</span><span class="op">]</span>,
  disease <span class="op">=</span> <span class="va">meta_data</span><span class="op">[[</span><span class="va">disease_column</span><span class="op">]</span><span class="op">]</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cells</span><span class="op">)</span>
<span class="co">#&gt; Registered S3 method overwritten by 'cli':</span>
<span class="co">#&gt;   method     from         </span>
<span class="co">#&gt;   print.boxx spatstat.geom</span>
<span class="co">#&gt; # A tibble: 6 × 3</span>
<span class="co">#&gt;   id                 sample disease</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;fct&gt;  &lt;fct&gt;  </span>
<span class="co">#&gt; 1 AAACCTGAGACACTAA_1 HC1    N      </span>
<span class="co">#&gt; 2 AAACCTGAGGAGTACC_1 HC1    N      </span>
<span class="co">#&gt; 3 AAACCTGAGGATATAC_1 HC1    N      </span>
<span class="co">#&gt; 4 AAACCTGAGGTCATCT_1 HC1    N      </span>
<span class="co">#&gt; 5 AAACCTGCACGGATAG_1 HC1    N      </span>
<span class="co">#&gt; 6 AAACCTGCAGGGAGAG_1 HC1    N</span></code></pre></div>
<p>Now lets check how many cells we have in each sample.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   HC1   HC2   HC3   HC4    O1    O2    O3    S1    C1    C2    C3    C4    C5 </span>
<span class="co">#&gt;  8466  8189  2566  2718  3542  3411   363 11872 17340  1292  1718  2071  2904</span></code></pre></div>
<p>We notice two things. The total number of cells is not at all equally distributed among all samples and also the authors seem to have mispelled some of the sample names. Let’s replace the “O” with an “S”.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cells</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span>, <span class="st">"O(?=[1-3])"</span>, <span class="st">"M"</span><span class="op">)</span>
<span class="va">cells</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span>, <span class="st">"S1"</span>, <span class="st">"S0"</span><span class="op">)</span>
<span class="va">cells</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span>, <span class="st">"^C(?=[1-5])"</span>, <span class="st">"S"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   HC1   HC2   HC3   HC4    M1    M2    M3    S0    S1    S2    S3    S4    S5 </span>
<span class="co">#&gt;  8466  8189  2566  2718  3542  3411   363 11872 17340  1292  1718  2071  2904</span></code></pre></div>
<p>So we actually have 4 samples as healthy control (HC), 3 samples from moderate (M) COVID-19 cases and 6 samples from severe cases (S). Based on the sample name we can add a condition column to the cells tibble.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cells</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span>, <span class="st">"\\d"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">condition</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    HC     M     S </span>
<span class="co">#&gt; 21939  7316 37197</span></code></pre></div>
<p>In the next step we load <code>lcsc</code> and compute the nearest neighbor graph per sample.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">lcsc</span><span class="op">)</span>

<span class="va">nn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_nn.html">run_nn</a></span><span class="op">(</span><span class="va">cells</span>, 
             <span class="va">pc_space</span>, 
             k<span class="op">=</span><span class="fl">50</span>,     <span class="co"># Compute 50 nearest neigbors per cell</span>
             dim<span class="op">=</span><span class="fl">30</span><span class="op">)</span>   <span class="co"># Using 30 PCs to compute NN</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">nn</span><span class="op">)</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ idx  : num [1:66452, 1:50] 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="co">#&gt;  $ dists: num [1:66452, 1:50] 0 0 0 0 0 0 0 0 0 0 ...</span></code></pre></div>
<p>Now we can load the application and start to annotate cells per sample. This requires some manual effort, but we do not rely on integration methods for multiple samples or unsupervised learning methods.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/lc_vis.html">lc_vis</a></span><span class="op">(</span>cells<span class="op">=</span><span class="va">cells</span>,
       counts<span class="op">=</span><span class="va">counts</span>,
       pc_space<span class="op">=</span><span class="va">pc_space</span>,
       embedding<span class="op">=</span><span class="va">embedding</span>,
       nn<span class="op">=</span><span class="va">nn</span>,
       k<span class="op">=</span><span class="fl">30</span> <span class="co"># Smoothing the expression over 50 nearest neighbors</span>
       <span class="op">)</span></code></pre></div>
<p>The classification of macrophages in each sample could look like this. We define macrophages as cells which express CD68 and do not express CD3E (T-cells) or TPPP3 (Epithelial cells).</p>
<p><img src="../man/figures/macrophages_application.png" width="1000"></p>
<p>To go back and refine the annotation one should save the annotation and the rule list.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/save.html">save</a></span><span class="op">(</span><span class="va">annotations</span>, <span class="va">rules</span>, file <span class="op">=</span> <span class="st">"macrophage_annotation.RData"</span><span class="op">)</span></code></pre></div>
<p>If one wishes to refine the annotation at some point, one can simply reload the application as done above and overwrite the <code>annotations</code> and <code>rules</code> objects by loading the <code>.RData</code> file.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"macrophage_annotation.RData"</span><span class="op">)</span></code></pre></div>
<p>Once we have annotated our cell type of interest we could perform for example differential gene expression analysis between the cells from the same cell type under different conditions (e. g. mild vs. severe COVID-19 infection). As we have only annotated macrophages so far, we will compare their expression profiles.</p>
<p>First, we will need to create pseudobulks for macrophages in each sample from the “M” (mild) and “S” (severe) samples.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We want to compare macrophages in mild vs. severe COVID-19</span>
<span class="va">mild_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"M"</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mild_samples</span><span class="op">)</span>
<span class="co">#&gt; [1] "M1" "M2" "M3"</span>
<span class="va">severe_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"S"</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">cells</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">severe_samples</span><span class="op">)</span>
<span class="co">#&gt; [1] "S1" "S0" "S2" "S3" "S4" "S5"</span>

<span class="va">cell_type</span> <span class="op">=</span> <span class="st">"Macrophages"</span>

<span class="va">mild_matrix</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">sample</span> <span class="kw">in</span> <span class="va">mild_samples</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">cell_type_barcodes</span> <span class="op">&lt;-</span> <span class="va">annotations</span><span class="op">[[</span><span class="va">sample</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="va">cell_type</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">barcode</span>
  <span class="va">mild_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">mild_matrix</span>, <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>, <span class="va">cell_type_barcodes</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">mild_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">mild_samples</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mild_matrix</span><span class="op">)</span>
<span class="co">#&gt;              M1   M2 M3</span>
<span class="co">#&gt; AL627309.1    9    8  0</span>
<span class="co">#&gt; AL669831.5  163  133  1</span>
<span class="co">#&gt; FAM87B        9    4  0</span>
<span class="co">#&gt; LINC00115   120  199 14</span>
<span class="co">#&gt; FAM41C       54   64  1</span>
<span class="co">#&gt; NOC2L      1430 1292 53</span>

<span class="va">severe_matrix</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">sample</span> <span class="kw">in</span> <span class="va">severe_samples</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">cell_type_barcodes</span> <span class="op">&lt;-</span> <span class="va">annotations</span><span class="op">[[</span><span class="va">sample</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="va">cell_type</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">barcode</span>
  <span class="va">severe_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">severe_matrix</span>, <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>, <span class="va">cell_type_barcodes</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">severe_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">severe_samples</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">severe_matrix</span><span class="op">)</span>
<span class="co">#&gt;              S1  S0  S2  S3  S4  S5</span>
<span class="co">#&gt; AL627309.1   30  10   4   6   6   3</span>
<span class="co">#&gt; AL669831.5  174 114  40  32  61  70</span>
<span class="co">#&gt; FAM87B       11   0   0   0   0   0</span>
<span class="co">#&gt; LINC00115   375 132  29  59  37  65</span>
<span class="co">#&gt; FAM41C       86  82  22  50  37  44</span>
<span class="co">#&gt; NOC2L      2357 787 147 563 296 772</span></code></pre></div>
<p>As we will be using <code>DESeq2</code> for our differential gene epxression analysis, we have to create a small dataframe describing the samples (“coldata”).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>sample<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mild_samples</span>, <span class="va">severe_samples</span><span class="op">)</span><span class="op">)</span>
<span class="va">coldata</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html">startsWith</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">$</span><span class="va">sample</span>, <span class="st">"M"</span><span class="op">)</span>, <span class="st">"mild"</span>, <span class="st">"severe"</span><span class="op">)</span>
<span class="va">coldata</span>
<span class="co">#&gt;   sample condition</span>
<span class="co">#&gt; 1     M1      mild</span>
<span class="co">#&gt; 2     M2      mild</span>
<span class="co">#&gt; 3     M3      mild</span>
<span class="co">#&gt; 4     S1    severe</span>
<span class="co">#&gt; 5     S0    severe</span>
<span class="co">#&gt; 6     S2    severe</span>
<span class="co">#&gt; 7     S3    severe</span>
<span class="co">#&gt; 8     S4    severe</span>
<span class="co">#&gt; 9     S5    severe</span></code></pre></div>
<p>Now we can simply run <code>DESeqDataSetFromMatrix</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2">DESeq2</a></span><span class="op">)</span><span class="op">)</span>
<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSetFromMatrix</a></span><span class="op">(</span>countData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">mild_matrix</span>, <span class="va">severe_matrix</span><span class="op">)</span>,
                                colData <span class="op">=</span> <span class="va">coldata</span>,
                                design <span class="op">=</span> <span class="op">~</span> <span class="va">condition</span><span class="op">)</span>
<span class="co">#&gt; converting counts to integer mode</span>
<span class="co">#&gt; Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in</span>
<span class="co">#&gt; design formula are characters, converting to factors</span>

<span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span>
<span class="co">#&gt; estimating size factors</span>
<span class="co">#&gt; estimating dispersions</span>
<span class="co">#&gt; gene-wise dispersion estimates</span>
<span class="co">#&gt; mean-dispersion relationship</span>
<span class="co">#&gt; final dispersion estimates</span>
<span class="co">#&gt; fitting model and testing</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></code></pre></div>
<p>And we could for example look at the genes assocatiated with the lowest adjusted p-values.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span><span class="op">)</span>
<span class="va">res</span> <span class="op">%&gt;%</span>
  <span class="va">as.data.frame</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">pvalue</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_head</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt;             baseMean log2FoldChange    lfcSE     stat         padj</span>
<span class="co">#&gt; IGKV3-11   4567.8948       28.74300 3.434250 8.369512 4.657479e-14</span>
<span class="co">#&gt; IGHV5-10-1 2336.6273       27.93190 3.434259 8.133311 2.832888e-13</span>
<span class="co">#&gt; IGKV1-9    1319.6604       27.13907 3.063320 8.859365 8.042479e-16</span>
<span class="co">#&gt; IGLV3-27   1222.8432       27.13060 3.434277 7.899948 1.394296e-12</span>
<span class="co">#&gt; IGHV1-46   1329.1049       26.95968 3.434274 7.850184 1.972134e-12</span>
<span class="co">#&gt; IGLV4-69   1046.2495       26.91493 3.434283 7.837134 2.084026e-12</span>
<span class="co">#&gt; IGHV3-74    973.1023       26.81567 3.434286 7.808223 2.541783e-12</span>
<span class="co">#&gt; IGLV2-23    814.9891       26.56442 3.434295 7.735043 4.005831e-12</span>
<span class="co">#&gt; IGKV1-6     777.4242       26.40558 3.434297 7.688786 5.641161e-12</span>
<span class="co">#&gt; IGHV1-24    655.1797       26.26125 3.434308 7.646738 7.385729e-12</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Philipp SL Schäfer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
